app/src/org/odk/collect/android/logic/ArchivedFormRemoteRestore.java

```diff
@@ -0,0 +1,131 @@
+package org.odk.collect.android.logic;
+
+import android.widget.Toast;
+
+import org.commcare.android.database.user.models.User;
+import org.commcare.android.framework.CommCareActivity;
+import org.commcare.android.tasks.DataPullTask;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.android.tasks.templates.CommCareTaskConnector;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.activities.FormRecordListActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Only used for developer debugging.
+ *
+ * Load saved form instances manually from a xml payload.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormRemoteRestore<A extends CommCareActivity> {
```

Comment:
I'm guessing I'm missing something nuanced here, but this class seems overly complex. Why does it need to use generics for the Activity type, since there's only one Activity that ever uses it? Worth nothing that if that isn't in fact necessary, you could simplify the class into just a single static method that lives in ArchivedFormManagement.java, which would be nice for having all of the archived form stuff in one place.